spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -883,6 +916,23 @@ private void checkIdle() {
 			}
 		}
 
+		private void idleBetweenPollIfNecessary() {
+			long idleBetweenPolls = this.containerProperties.getIdleBetweenPolls();
+			if (idleBetweenPolls > 0) {
+				idleBetweenPolls = Math.min(idleBetweenPolls,
+						this.maxPollInterval - (System.currentTimeMillis() - this.lastPoll));
```

Comment:
After further though, I think we need some "headroom" here - otherwise if we wait for all of the remaining time in `maxPollInterval`, we'll be in a race with the broker regarding a rebalance.

Maybe we should consider the minimum to no more than several seconds (10?) less than the remaining time before a rebalance will occur.